CONTRIBUTING.md

```diff
@@ -0,0 +1,143 @@
+
+Provide more context by answering these questions:
+* **Did the problem start happening recently** (e.g. after updating to a new version of Gateway) or was this always a problem?
+* If the problem started happening recently, **can You reproduce the problem in an older version of Gateway?** What's the most recent version in which the problem doesn't happen?
+
+
+* **Can You reliably reproduce the issue?** If not, provide details about how often the problem happens and under which conditions it normally happens.
+
+Include details about Your configuration and environment:
+
+* **Which released version are You using?** You could find it in a special file.
+* **What is the name and version of the OS You are using**?
+
+### Suggesting Enhancements
+
+Before creating enhancement suggestions, please check the repository issue list as You might find out that You don't need to create one. When You are creating an enhancement suggestion, please include as many details as possible. Fill in [the template](TEMPLATES/feature_request.md), including the steps that You imagine You would take if the feature You're requesting existed.
+Explain the feature and include additional details:
+
+* **Use a clear and descriptive title** for the issue to quickly figure out the idea of suggestion.
+* **Provide a step-by-step description of the suggested enhancement** in as many details as possible.
+* **Describe the current behaviour** and **explain which behaviour You expected to see instead** and why.
+* **Explain why this enhancement would be useful**
+* **List some other applications where this enhancement exists.**
+* **Specify which version of Gateway You're using.** You can get a version from special config file
+* **Specify the name and version of the OS You are using.**
+
+### Pull Requests
+
+Here the checklist of what You need to create a successful pull request:
+1. Point issue on the repository
+2. Descriptive commit messages. [Checkout the guide](#git-commit-messages)
+3. In case You have changed the behaviour of some feature/ widely used function, etc....
+     1. Provide a justification for this change
+     2. Change according documentation that listed in README in the folder with code.
+4. There is no need to assign reviewers to Your PR, we will do it for You.
+5. Pay attention to reviewers suggestions.
+6. Pull request will merged if and only if there is 3 approvals on it
+
+### Documentation Maintenance
+
+Firstly, [checkout out the documentation guide](#documentation-styleguide).
+
+### Styleguides
+
+* [Google C++ styleguide](https://google.github.io/styleguide/cppguide.html)
+* [Documentation Styleguide](#documentation-styleguide)
+* [Git Commit Messages](#git-commit-messages)
+
+### Git Commit Messages
+
+* [Use this guide to write acceptable commit message](https://wiki.openstack.org/wiki/GitCommitMessages)
+
+### Documentation Styleguide
+
+
+### Additional notes
+
+
+#### Thanks
```

Comment:
It`s better to say "gratitude" or "acknowledgment" :-)

Reply:
Not sure regarding "acknowledgment", I guess You have pointed this word to the "Additional notes" phrase?